### ISSUE TITLE
Generate Windows executable on release tag push

### ISSUE THREAD

**Issue by okaneco** [okaneco/rscolorq#12]
> Dear @okaneco,
Could you be so kind to generate ```.exe``` for the rest of us who are mere Windows users w/o compiler?
> _Originally posted by @sergeevabc in https://github.com/okaneco/rscolorq/issues/2#issuecomment-802295509_

I'd be interested in doing this, however I don't know how to configure GH Actions to build a Rust binary for release.

**Comment by ksmoore17** [comment]
I found [this](https://alican.codes/rust-github-actions/) and [this](https://zork.net/~st/jottings/rust-windows-and-debian.html ) relevant to the problem. I'm going to try to figure it out.

**Comment by okaneco** [comment]
Those look helpful, I found this as well. Some combination of the workflows is probably what we want.

https://github.com/Rigellute/spotify-tui/blob/master/.github/workflows/cd.yml  
https://github.com/rust-github/template/blob/master/.github/workflows/cd.yml (from this video https://youtu.be/meAz2RQMHvM?t=1331, he credits spotify-tui project)

**Comment by ksmoore17** [comment]
Nice one, a combination of those two worked, thanks. Tried to stick closer to the template one because it was less specific to spotify ~~but it also just had a ton of syntax errors~~. All in all it wasn't very painful, and I was able to test with a release in my fork: 5 binaries - Windows, 3 linux arches, and mac - all built and published to the github release successfully. I didn't bother to set up a crates api key.

edit: The executable on windows (when double click to open) flashes a console window for a moment then closes. This also happened when I tried to run the executable created by a cargo build. To be honest I don't know what to expect from the exe or how rust "main" works. Running the exe from command line does work! I forget if there is any feedback from the program when processing happens successfully, but I can't see any.

edit edit: The "syntax errors" that I mentioned are not syntax errors but so that that repo can actually be used as a template: https://github.com/cobalt-org/liquid-rust

`.\rscolorq.exe --input input.png --output output.png`
